From 05290131812f2faf548ab76124cef53689053338 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Tue, 29 Nov 2022 10:35:28 +0200
Subject: [PATCH] [SW-0] greco: init full 64K spsram

Previously we were initializing spsram after 4K offset, as arcboot rom
copies info like keys, info for image verification, security efuse info.
Since security is not POR for greco, we don't use this info.
Instead of 60K and leaving first 4K offset, now we init full 64K of
spsram as we use spsram for mailbox, bootinfo, rodata etc.

Change-Id: Ib07d4b4db10801ebd99cbecaae22634e19c0ecab
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index 56d61ac..07b4fa0 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -169,7 +169,7 @@
 		 PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H_SECURITY_BYPASS_MASK);
 }
 
-#define SP_SRAM_60K_SZ_16B_GRANULARITY	0xF00
+#define SP_SRAM_64K_SZ_16B_GRANULARITY	0x1000
 void preboot_sp_sram_init(void)
 {
 	u32 mem_cfg, cpy_sts, rst_src;
@@ -184,18 +184,16 @@
 	if (rst_src != HL_COLD_RST)
 		return;
 
-	/* Use memblock to initialize scratchpad memory from 4K offset
-	 * as first 4K offset is used by arcboot rom
-	 */
-	regmap_writel((u32)(HL_SP_SRAM_BASE + SZ_4K),
+	/* Use memblock to initialize scratchpad memory */
+	regmap_writel((u32)(HL_SP_SRAM_BASE),
 			mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_L);
-	regmap_writel((u32)((HL_SP_SRAM_BASE + SZ_4K) >> 32),
+	regmap_writel((u32)((HL_SP_SRAM_BASE) >> 32),
 			mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H);
 
 	mem_cfg = regmap_readl(mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
 	mem_cfg &= ~(PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK);
-	/* set size of 60k in granuality of 16 bytes */
-	mem_cfg |= (SP_SRAM_60K_SZ_16B_GRANULARITY & PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK);
+	/* set size of 64k in granuality of 16 bytes */
+	mem_cfg |= (SP_SRAM_64K_SZ_16B_GRANULARITY & PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK);
 	mem_cfg |= (BIT(PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_SHIFT));
 
 	regmap_writel(mem_cfg, mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
